ibm-watson-workitem/src/main/java/org/jbpm/process/workitem/ibm/watson/ClassifyImageWorkitemHandler.java

```diff
@@ -0,0 +1,132 @@
+                @WidMavenDepends(group = "com.ibm.watson.developer_cloud", artifact = "java-sdk", version = "4.0.0")
+        })
+public class ClassifyImageWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClassifyImageWorkitemHandler.class);
+    private static final String RESULT_VALUE = "Classification";
+    private VisualRecognition service;
+
+    private String apiKey;
+
+    public ClassifyImageWorkitemHandler(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+
+        Document classificationImage = (Document) workItem.getParameter("ImageToClassify");
+
+        Map<String, Object> widResults = new HashMap<String, Object>();
+
+        if (classificationImage != null) {
+
+            try {
+                if(service == null) {
+                    VisualRecognition service = new VisualRecognition(VisualRecognition.VERSION_DATE_2016_05_20);
+                    service.setApiKey(apiKey);
+                }
+
+                ByteArrayInputStream imageStream = new ByteArrayInputStream(classificationImage.getContent());
+
+                ClassifyOptions classifyOptions = new ClassifyOptions.Builder()
+                        .imagesFile(imageStream)
+                        .imagesFilename(classificationImage.getName())
+                        .parameters("{\"owners\": [\"me\"]}")
+                        .build();
+                ClassifiedImage result = service.classify(classifyOptions).execute().getImages().get(0);
+                if (result.getError() != null) {
+                    ErrorInfo errorInfo = result.getError();
+                    logger.error("Error classifying image: " + errorInfo.getDescription());
+                    workItemManager.completeWorkItem(workItem.getId(),
+                                                     null);
+                } else {
+                    List<ClassifierResult> classificationResults = result.getClassifiers();
+                    for (ClassifierResult classification : classificationResults) {
+                        Map<String, String> resultToAdd = new HashMap<>();
+
+                        for (ClassResult classResult : classification.getClasses()) {
+                            resultToAdd.put("classificationname",
+                                            classification.getName());
+                            resultToAdd.put("classname",
+                                            classResult.getClassName());
+                            resultToAdd.put("score",
+                                            String.valueOf(classResult.getScore()));
+                            resultToAdd.put("typehierarchy",
+                                            classResult.getTypeHierarchy());
+                        }
+
+                        widResults.put(RESULT_VALUE,
```

Comment:
maybe have 4 results objects (classification name, classname, score, type hierarchy) instead of adding them into the a map and then returning map, will be more difficult to deal with map as result down the road in the process.